lightning/src/ln/channelmanager.rs

```diff
@@ -2966,6 +2966,52 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 			Err(e) => { Err(APIError::APIMisuseError { err: e.err })}
 		}
 	}
+
+	/// Process pending events from the ManyChannelMonitor.
+	fn process_pending_monitor_events(&self) {
+		let mut failed_channels = Vec::new();
+		// TODO: Event release to users and serialization is currently race-y: it's very easy for a
+		// user to serialize a ChannelManager with pending events in it and lose those events on
+		// restart. This is doubly true for the fail/fulfill-backs from monitor events!
+		{
+			//TODO: This behavior should be documented.
```

Comment:
I think this should be duplicated across the two get_and_clear* methods - the point is that the behavior where we query channelmonitors when clearing other events is, at the minimum, very surprising.

Reply:
I think we move it out of `get_and_clear_*` into the top-level API ? Did we have a strong reason to place the monitor query other than not complexifying API, can't remember ?

You may want to call monitor only after each block connection as both events `CommitmentTxBroadcasted/HTLCEvent` are only updated after a block which is not the case of `Event/MessageSendEvent`

Reply:
Yea, there's no good reason for it aside from not complexifying the current API - adding a new method which users have to poll is terrible, and we can't rely on users calling ChannelMonitor::block_connected before they call ChannelManger::block_connected (which would be required to poll it in block_connected). Not sure where else to put it.